packages/safe-core-sdk/src/Safe.ts

```diff
@@ -676,6 +676,45 @@ class Safe {
     return safeTransaction
   }
 
+  /**
+   * Checks if a Safe transaction can be executed successfully with no errors.
+   *
+   * @param safeTransaction - The Safe transaction to check
+   * @param options - The Safe transaction execution options. Optional
+   * @returns TRUE if the Safe transaction can be executed successfully with no errors
+   */
+  async isValidTransaction(
+    safeTransaction: SafeTransaction,
+    options?: TransactionOptions
+  ): Promise<boolean> {
+    const signedSafeTransaction = await this.createTransaction({
```

Comment:
The first part of this function which creates a copy of the transaction could be moved to some util function.